.github/workflows/BuildInfrastructure.yml

```diff
@@ -0,0 +1,43 @@
+﻿name: Build infrastructure
+
+on:
+  workflow_call:
+    secrets:
+      AZURE_CLIENT_ID:
+        required: true
+      AZURE_TENANT_ID:
+        required: true
+      AZURE_SUBSCRIPTION_ID:
+        required: true
+    inputs:
+      environment:
+        required: true
+        type: string
+
+jobs:
+  build:
+    name: Build infrastructure for ${{ inputs.environment }}
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    environment: ${{ inputs.environment }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: OIDC Login to Azure Public Cloud
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: Lint Bicep files
+        run: az bicep build --file .azure/main.bicep
```

Comment:
I følge [MS](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-cli#build):
>The build command converts a Bicep file to an Azure Resource Manager template (ARM template). Typically, you don't need to run this command because it runs automatically when you deploy a Bicep file. Run it manually when you want to see the ARM template JSON that is created from your Bicep file.

Hva er grunnen for at vi konverterer fra bicep til arm når det gjøres i deploy av bicep? 🙂

Reply:
jeg og @MagnusSandgren  tok det på huddle. 

Største reasoning er at vi kan enkelt verifisere at infrastruktur er lintet og bygger fint. Så vi kan "fail early". 

Dette kan bygges videre inn i en reusable action som kan brukes kun for den typen validering i caser der vi ikke nødvendigvis vil kjøre en preflight request eller en ren deploy